File: v_1.0.2/source/hin-common/src/main/java/com/hin/domain/vo/TaskVO.java
package com.hin.domain.vo;

import java.util.HashMap;
import java.util.Set;

public class TaskVO {
	public String taskId = "";
	public String messageID = "";
	public String message="";
	public Set<String> taskOutComes;
	public HashMap<String, String> assignablePeople;

	public String getTaskId() {
		return taskId;
	}

	public void setTaskId(String taskId) {
		this.taskId = taskId;
	}

	public String getMessageID() {
		return messageID;
	}

	public void setMessageID(String messageID) {
		this.messageID = messageID;
	}

	public Set<String> getTaskOutComes() {
		return taskOutComes;
	}

	public void setTaskOutComes(Set<String> taskOutComes) {
		this.taskOutComes = taskOutComes;
	}

	public HashMap<String, String> getAssignablePeople() {
		return assignablePeople;
	}

	public void setAssignablePeople(HashMap<String, String> assignablePeople) {
		this.assignablePeople = assignablePeople;
	}

	public String getMessage() {
		return message;
	}

	public void setMessage(String message) {
		this.message = message;
	}
}

File: v_1.0.2/source/hin-web/src/main/webapp/html/js/vo/TaskVO.js
var HIN;
if (!HIN)
	HIN = {};
/**
 * TaskVO
 *   
 * Represent the task information against a message, which will be
 * identify by workflow configuration based on the message type.
 */
HIN.TaskVO = function() {
	/** taskId holds the task id. */
	this.taskId = "";
	/** messageID holds the messageId of the task. */
	this.messageID = "";
	/**
	 * taskOutComes holds outcomes of the task which is dependend on the
	 * workflow task
	 */
	this.taskOutComes = [];
	/**
	 * assignablePeople holds assignablePeople of the task which is dependend on
	 * the workflow task
	 */
	this.assignablePeople = new HIN.HashMap();

	/** assignee holds the assignee of task. */
	this.assignee = "";
	/** outCome holds the outCome of task. */
	this.outCome = "";
	/** taskName holds the name of task. */
	this.taskName = "";
};

HIN.TaskVO.prototype.addOutCome = function(outcome) {
	this.taskOutComes.push(outcome);
};

HIN.TaskVO.prototype.addAssignablePeople = function(key, value) {
	this.assignablePeople.put(key, value);
};
/**
 * getOutComes method will return all outcomes under task.
 */
HIN.TaskVO.prototype.getOutComes = function() {
	return this.taskOutComes;
};
/**
 * getAssignablePeople method will return all assignees under task.
 */
HIN.TaskVO.prototype.getAssignablePeople = function(key) {
	var map = this.assignablePeople.get(key);
	if (map) {
		return map.value;
	}
	return null;
};

HIN.TaskVO.prototype.toString = function() {
	return "[" + this.taskId + " , " + this.messageID + " , " + " , "
			+ this.taskOutComes + ", " + this.assignablePeople + ","
			+ this.assignee + "," + this.outCome + "," + this.taskName + "]";
};
